File: CareConnect/CareConnect.Models/SearchObjects/ChildrenDiagnosisSearchObject.cs
using System;
using CareConnect.Models.SearchObjects;

namespace CareConnect.Models.SearchObjects
{
    public class ChildrenDiagnosisSearchObject : BaseSearchObject<ChildrenDiagnosisAdditionalData>
    {
        public DateTime? DiagnosisDateGTE { get; set; }

        public DateTime? DiagnosisDateLTE { get; set; }

        public string? ChildFirstNameGTE { get; set; } = null!;

        public string? ChildLastNameGTE { get; set; } = null!;

        public string? DiagnosisNameGTE { get; set; } = null!;
    }
}

File: UI/careconnect_mobile/lib/models/search_objects/children_diagnosis_search_object.dart
import 'package:careconnect_mobile/models/search_objects/children_diagnosis_additional_data.dart';
import 'package:json_annotation/json_annotation.dart';

part 'children_diagnosis_search_object.g.dart';

@JsonSerializable(explicitToJson: true)
class ChildrenDiagnosisSearchObject {
  String? fts;
  DateTime? diagnosisDateGTE;
  DateTime? diagnosisDateLTE;
  String? childFirstNameGTE;
  String? childLastNameGTE;
  String? diagnosisNameGTE;
  int? page;
  String? sortBy;
  bool? sortAscending;
  ChildrenDiagnosisAdditionalData? additionalData;
  bool? includeTotalCount;

  ChildrenDiagnosisSearchObject({
    this.fts,
    this.diagnosisDateGTE,
    this.diagnosisDateLTE,
    this.childFirstNameGTE,
    this.childLastNameGTE,
    this.diagnosisNameGTE,
    this.page,
    this.sortBy,
    this.sortAscending,
    this.additionalData,
    this.includeTotalCount,
  });

  factory ChildrenDiagnosisSearchObject.fromJson(Map<String, dynamic> json) =>
      _$ChildrenDiagnosisSearchObjectFromJson(json);

  Map<String, dynamic> toJson() => _$ChildrenDiagnosisSearchObjectToJson(this);
}
